### ISSUE TITLE
Registry Recipe book

### ISSUE THREAD

**Issue by RichardScothern** [docker/distribution#879]
Some HOWTOs in the registry would be useful for common tasks.  The knowledge is spread around various .md files.

@dmp42 : if you can specify what form a recipe should take we can split up the work

 [ ] Setup for ELB, Apache, Nginx
 [ ] no-auth pull with auth push setup
 [ ] load balancing and scaling tips and tricks
 [ ] read-only registry
 [ ] proxy cache registry (?)

**Comment by dmp42** [comment]
A recipe could be a simple separate md doc, similar to `nginx.md` (https://github.com/docker/distribution/blob/master/docs/nginx.md)

The recipe should:

 * spell out a use case "if you want to achieve Z, this is for you"
 * state a clear objective: "(achieving Z) we are going to do B with A"
 * mention alternatives: "you could also do C, and that's more recommended"
 * be complete: running the provided scripts and instructions by the letter should work out of the box

Recipes should not be;

 * linked directly from the top level docs navigation

but may all be listed in a "recipe index" document.

What do you all think? @docker/distribution-maintainers

**Comment by stevvooe** [comment]
This sounds okay. Maintaining these can be a burden if there are differences from version to version. As long as a warning about these being a guide and not a panacea should be sufficient.

@kelseyhightower had asked about the auth push/open pull at some point. I believe he got it working.

**Comment by dmp42** [comment]
Agreed - need to limit the maintenance burden to an absolute minimum.
These recipes should be based on fixed versions docker images whenever possible (apache, nginx).

Also clearly recipes should be "not immediately visible" and somewhat "buried", with clear mentions about their status.

**Comment by thaJeztah** [comment]
Would the example for setting up a hybrid v1/v2 registry (that was originally on the deploy page) be a candidate for a recipe? I'm actually not sure that's something that's still "supported" / "recommended", but wanted to point a user to that example recently, but forgot it was no longer there :smile:
